### PR TITLE
Aut16 windows

### PR DESCRIPTION
**Pull request by somanianshul**
CHEF Base Cookbook Port to windows:

Integration and Unit test Status: Passing for Amazon, RedHat, SUSE & Windows 

- Added Config.xml file to disable automatic change of hostname in windows in EC2 Config.
- Install chocolatey package manager in windows system
- Uses chocolatey to install AWS CLI
- Checks of the host is windows and runs windows centric recipes and tests (integration and unit)
- Changed the windows reboot attribute to false to facilitate the integration testing from kitchen without rebooting the instance
- Does not install Python, mLocate, Vim and Docker if the host is windows. 

TODO - Code review 

FUTURE TASKS - add functionality to add firewall rules to allow ports for PowerShell Remoting in Windows.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -4,8 +4,13 @@
 #
 # Copyright (c) 2016 Soltius New Zealand Ltd, All Rights Reserved.
 
-include_recipe 'chef-base::config'
-include_recipe 'chef-base::docker'
+if node['platform'] != 'windows'
```
Comment (igorlg) [inline recipes/default.rb]: @somanianshul invert the logic here. Use something like 
`if node['platform'] == 'windows'
  include_recipe 'chocolatey::default'
else
  ...
end`

File: spec/unit/recipes/awscli_spec.rb [hunk 2]
```diff
@@ -11,8 +11,14 @@
     ChefSpec::ServerRunner.new(PLATFORM).converge(described_recipe)
   end
 
-  it 'installs python using system packages' do
-    expect(chef_run).to install_python_runtime('2').with(provider: PoisePython::PythonProviders::System)
+  if PLATFORM['platform'] != 'windows'
```
Comment (igorlg) [inline spec/unit/recipes/awscli_spec.rb]: Also invert the logic here.

File: spec/unit/recipes/default_spec.rb [hunk 3]
```diff
@@ -15,22 +15,28 @@
     expect { chef_run }.to_not raise_error
   end
 
-  it 'includes recipe chef-base::docker' do
-    expect(chef_run).to include_recipe('chef-base::docker')
+  if PLATFORM['platform'] != 'windows'
```
Comment (igorlg) [inline spec/unit/recipes/default_spec.rb]: Also invert the logic here.

File: spec/unit/recipes/docker_spec.rb [hunk 4]
```diff
@@ -6,20 +6,22 @@
 
 require 'spec_helper'
 
-describe 'chef-base::docker' do
-  let(:chef_run) do
-    ChefSpec::ServerRunner.new(PLATFORM).converge(described_recipe)
-  end
+if PLATFORM != %w(windows 2012r2)
```
Comment (igorlg) [inline spec/unit/recipes/docker_spec.rb]: This if won't ever be true. PLATFORM is a hash `{platform: 'xxxx', version: 'yyy'}`

File: test/integration/default/serverspec/default_spec.rb [hunk 5]
```diff
@@ -1,62 +1,67 @@
 require 'spec_helper'
 
 # TODO: Add support for Amazon Linux
-describe 'docker installation' do
-  DOCKER_PKG = os[:family] == 'redhat' ? 'docker-engine' : 'docker'
+describe 'linux servespec tests' do
+  if os[:family] != 'windows'
```
Comment (igorlg) [inline test/integration/default/serverspec/default_spec.rb]: Invert logic here.

File: recipes/default.rb (hunk 1 above)
Comment (somanianshul) [inline recipes/default.rb]: done

File: spec/unit/recipes/awscli_spec.rb (hunk 2 above)
Comment (somanianshul) [inline spec/unit/recipes/awscli_spec.rb]: done

File: spec/unit/recipes/default_spec.rb (hunk 3 above)
Comment (somanianshul) [inline spec/unit/recipes/default_spec.rb]: done

File: spec/unit/recipes/docker_spec.rb (hunk 4 above)
Comment (somanianshul) [inline spec/unit/recipes/docker_spec.rb]: rectified the conditional statement

File: test/integration/default/serverspec/default_spec.rb (hunk 5 above)
Comment (somanianshul) [inline test/integration/default/serverspec/default_spec.rb]: there is no if else condition hence the conditional statement cannot be inverted.

**Comment by igorlg** [comment]
INTEGRATION TEST FAILING ON RHEL DUE TO CHEF CDN ISSUE.
Will retry jobs later.

**Comment by igorlg** [comment]
Tests passed on RHEL - Problem with CDN fixed by Chef.

**Pull request merged**